<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
  <head>
    <p><a id="logo" href="http://scala-lang.org/"><img src="/sites/default/files/icons/Scala_Logo2008.png" alt="scala-lang.org" border=0/></a></p>
    <title id="Title">Aladdin - Scala Bugtracking</title>
        <style type="text/css">
        table {
            font-size: 10pt;
            background-color: #ebb85c;
        }
        tr.impair        { background-color: #f0be64 }
        tr.impairContrib { background-color: #f0be64 }

#buraq-content {
	float: none;
	color: black;
	background-color: white;
	font-size: 1.25em;
	margin-left: 9.6em;
}
        </style>
      </head>
  <body><!--DisplayBugsBlock_Start-->




<table WIDTH="100%" cellspacing="4" cellpadding="2">

	<tr><td></td>
		<th>[#898]</th>
		<th><strong>project:</strong> compiler</th>
		<th><strong>priority:</strong> low</th>
		<th><strong>category:</strong> bug</th>
        </tr>

        <tr>
          <td></td>
	  <th align="center">submitter</th>
	  <th align="center">assigned to</th>
	  <th align="center">status</th>
	  <th align="center">date submitted</th>
        </tr>

        <tr>
          <td></td>
	  <td align="center" bgcolor="#FFFFFF">Sean</td>
	  <td align="center" bgcolor="#FFFFFF">Martin</td>
	  <td align="center" bgcolor="#FFFFFF">fixed</td>
          <td align="center" bgcolor="#FFFFFF">2007-01-17 14:04:37.0</td>

	</tr>

        <tr>
		<th align="right">subject</th>
		<td colspan="4" bgcolor="#FFFFFF">verify error when creating nested class instance during constructor call</td>
	</tr>
	<tr>	
		<th align="right">code</th>
		<td colspan="4" bgcolor="#FFFFFF">
	<font size="-1"><pre>package test;
object Test {
  abstract class SortedSetWrapper[A] {
    def underlying : java.util.TreeSet;
  }
  class TreeSet[A &lt;% Ordered[A]](val underlying : java.util.TreeSet) extends SortedSetWrapper[A] {
    private class Comparator extends java.util.Comparator {
      def compare(x0 : Any, x1 : Any) = 
        x0.asInstanceOf[A].compare(x1.asInstanceOf[A]);
    }
    def this() = this(new java.util.TreeSet(new Comparator));
  }
  def main(args : Array[String]) : Unit = 
    new TreeSet[String];
}
</pre></font></td>
	</tr>
	<tr>
		<th style="text-align:right;">what happened</th>
		<td colspan="4" style="background-color:#FFFFFF;" ><pre>n execute: <pre>Exception in thread "main" java.lang.VerifyError: (class: test/Test$TreeSet, method: <init> sign\
ature: (Lscala/Function1;)V) Expecting to find object/array on stack
	at test.Test$.main(Test.scala:18)
	at test.Test.main(Test.scala)</pre>
</pre></td>
	</tr>
	<tr>
		<th align="right">what expected</th>
		<td colspan="4" bgcolor="#FFFFFF" >A compiler error.</td>
	</tr>
	<tr>
	
  <td colspan="4" align="right">
      
      <a href="displayBugs.do.html">
        [back to overview]</a></td></tr>

</table>

  <tr>
          <td align="center" colspan="2">
            <strong>Changes of this bug report</strong>
          </td>
        </tr><table><tr>
        <td>
        <strong>Iulian&nbsp; edited on &nbsp;2007-01-22 11:21:11.0</strong>
        </td>
        </tr>
<tr><td bgcolor='#FFFFFF'>
This is very similar to bug #908: the call to 'new Comparator' as argument to another constructor is illegal since it needs an outer instance, which is really the object being constructed (uninitialized). As bug #908, this exists only in the svn repository (the current release points out the error).
</td></tr>
<tr>
        <td>
        <strong>Martin&nbsp; edited on &nbsp;2007-01-23 16:18:33.0</strong>
        </td>
        </tr>
<tr><td bgcolor='#FFFFFF'>
This is a duplicate of 908.
</td></tr></table>
<!--DisplayBugsBlock_End-->  </body>
</html>